src/eos-dropbox-app

```diff
@@ -198,10 +199,49 @@ class DropboxLauncher():
             logging.info("Not the main launcher instance. Exiting")
             self._mainloop.quit()
 
+    def _launch_dropbox(self):
+        self._launch_dropbox_daemon()
+        self._setup_config_monitor()
+
+    def _setup_config_monitor(self):
+        config = Gio.File.new_for_path(os.path.expanduser(os.path.expanduser(DROPBOX_CONFIG)))
+        self._config_monitor = config.monitor(Gio.FileMonitorFlags.NONE)
+        self._config_monitor.connect('changed', self._on_config_changed)
```

Comment:
As discussed on Slack I left this behavior on purpose because if the Dropbox folder is deleted for some reason, the daemon shows a dialog for the user to relink it again, in which case I believe that it's a better UX to show the directory after the user sets it up again (otherwise it's just a click in a browser window with no apparent feedback in the system).
Also, this shouldn't be happening too much so between the two options I chose the one that gives a better feedback to the user as that's what I expect as one.